fix build issues related to Open-MPI function deletion

jeffhammond opened the pull request:
Open-MPI implemented the deletion of MPI_Errhandler_* functions,
including helpful notification that causes the build to fail.
This change disables the missing functions.

Signed-off-by: Hammond, Jeff R <jeff.r.hammond@intel.com>

Resolves https://github.com/pmodels/casper/issues/36

minsii reviewed (commented):
The commit looks good to me. The travis failures are irrelevant to this PR. I will look into them. Before merge, can you please clean up the unrelated changes in file copyright comment?

jeffhammond commented:
@minsii Okay, I did that.  I just know it was impossible for that file to have been created in Casper in 2001 😉

Commit:
fix build issues related to Open-MPI function deletion

Open-MPI implemented the deletion of MPI_Errhandler_* functions,
including helpful notification that causes the build to fail.
This change disables the missing functions.

Resolves https://github.com/pmodels/casper/issues/36

Signed-off-by: Hammond, Jeff R jeff.r.hammond@intel.com

minsii commented:
> @minsii Okay, I did that. I just know it was impossible for that file to have been created in Casper in 2001 😉

Yes, that was a mistake...Good catch! I will do a copyright cleanup for all files in a separate commit.
